juju/framework.py

```diff
@@ -189,13 +194,8 @@ def __init__(self, parent, key):
         # TODO This can probably be dropped, because the event type is only
         # really relevant if someone is either emitting the event or observing
         # it.
-        for cls in type(self).__mro__:
-            for attr_name, attr_value in cls.__dict__.items():
-                if isinstance(attr_value, Event):
-                    event_type = attr_value.event_type
-                    event_kind = attr_name
-                    emitter = self
-                    self.framework.register_type(event_type, emitter, event_kind)
```

Comment:
I don't want to sound picky, but this logic seems significantly clearer than what is on the right-hand side, which is more valuable win than shortness. The change to use getmembers sounds good (again, because it makes it more clear), but can we please preserve the logic above?  The getmembers call should replace just the loop itself I think.